### PR TITLE
Checkpoint FWv2: Improved login mechanism

### PR DESCRIPTION
**Pull request by dorschw**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39704
fixes: https://github.com/demisto/etc/issues/39369

## Description
Login-logout mechanism improved, made clearer
More logs on login/logout/restore SID

## Minimum version of Cortex XSOAR
- [x] 5.0.0

## Does it break backward compatibility?
 - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py [hunk 1]
```diff
@@ -67,7 +75,7 @@ def test_connection(self):
         Otherwise, an exception should be raised by self._http_request()
         """
         response = self._http_request(method='POST', url_suffix='show-api-versions', headers=self.headers,
-                                      ok_codes=(200, 500), resp_type='response')  # returns requests.Response
+                                      ok_codes=(200, 500), resp_type='response', json_data={})
```
Comment (DeanArbel) [inline Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py:78]: why did you add `json_data` did you intend to pass it to `self._http_request`?

File: Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py [hunk 2]
```diff
@@ -1961,7 +1970,9 @@ def main():
         else:
             raise NotImplementedError(f"Unknown command {demisto.command()}.")
 
-        if not stay_logged_in:
+        if client.has_performed_login:
+            # this party is not reached when login() is explicitly called
```
Comment (DeanArbel) [inline Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py:1974]: party! 🎂 
```suggestion
            # this part is not reached when login() is explicitly called
```

**Review by DeanArbel — APPROVED** [review]
Nice

File: Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py (hunk 1 above)
Comment (dorschw) [reply Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py:78]: Similar to the issue we had with `login()`, using a json that isn't explicitly empty seems to be considered invalid on the FW side.

File: Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py (hunk 2 above)
Comment (dorschw) [reply Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py:1974]: ain't no party like a FWv2 party

File: Packs/CheckpointFirewall/ReleaseNotes/2_1_1.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### CheckPoint Firewall v2
+- Solved an issue where logout would be preformed after every command, causing problems in Playbooks.
+- Solved an issue where provided `server` values starting with `https://` would not work.
```
Comment (DeanArbel) [inline Packs/CheckpointFirewall/ReleaseNotes/2_1_1.md:5]:
```suggestion
- Fixed an issue where logout would be performed after every command execution.
- Fixed an issue where provided `server` values starting with `https://` would not work.
```

**Review by DeanArbel — APPROVED** [review]
